pkg/gofr/logging/dynamicLevelLogger.go

```diff
@@ -0,0 +1,89 @@
+package logging
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"time"
+
+	"gofr.dev/pkg/gofr/service"
+)
+
+const (
+	levelFetchInterval = 15 * time.Second
+	requestTimeout     = 5 * time.Second
+)
+
+func NewRemoteLogger(level Level, remoteConfigURL string) Logger {
+	l := remoteLogger{
+		remoteURL:    remoteConfigURL,
+		Logger:       NewLogger(level),
+		currentLevel: level,
+	}
+
+	if remoteConfigURL != "" {
+		go l.UpdateLogLevel()
+	}
+
+	return l
+}
+
+type remoteLogger struct {
+	remoteURL    string
+	currentLevel Level
+	Logger
+}
+
+func (r *remoteLogger) UpdateLogLevel() {
+	ticker := time.NewTicker(levelFetchInterval)
+	defer ticker.Stop()
+
+	remoteService := service.NewHTTPService(r.remoteURL, r.Logger)
+
+	for range ticker.C {
+		newLevel, err := fetchAndUpdateLogLevel(remoteService)
+		if err == nil {
+			r.changeLevel(newLevel)
+
+			if r.currentLevel != newLevel {
+				r.Infof("LOG_LEVEL updated from %v to %v", r.currentLevel, newLevel)
+				r.currentLevel = newLevel
+			}
+		}
+	}
+}
+
+func fetchAndUpdateLogLevel(remoteService service.HTTP) (Level, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout) // Set timeout for 5 seconds
+	defer cancel()
+
+	resp, err := remoteService.Get(ctx, "", nil)
+	if err != nil {
+		return INFO, err
+	}
```

Comment:
If remote service does not report anything because of error, we should not be returning "INFO". Because if the previous level was anything else (like WARN), we will end up changing to INFO which is not the intention - as the long level should not change in such case. @Umang01-hash

Reply:
Sure @vikash. Understood your point. We can return the current log Level in such case. Will do the alteration